### DISCUSSION TITLE
Danke für deine schnelle Antwort :-)

### DISCUSSION THREAD

**Discussion by Tiese** [creepystefan/esphome-XY6020#2] [Q&A]
Hallo @creepystefan,

> Hey @Tiese
Das sinilink wifi Modul kann wie ein normaler ESP benutzt werden.
Am besten schaust du auf meiner Seite. Dort habe ich die GPIOs
Und programmieranschlüsse angezeigt.
In meinem Beispiel wird das Modul mit ESPHome bespielt.
Und kann alles gemacht werden.
Allerdings muss ich noch den Code für das sinilink mit dem XY6020 anpassen. Momentan ist es für einen ESP32 angepasst.
Bitte schreibe mir doch direkt auf meiner GitHub Seite.
API wird über ESPhome benutzt
Auch eine Web Oberfläche.
Das Sinilink Portal wird nicht gebraucht.

Dann versuche ich es mal hier. ;-)

Wie kann ich auf den Sinilink Daten aufspielen?
Ein USB-Anschluss hat es ja nicht :-/

Ich danke dir für deine Unterstützung!

Schöne Grüße
Tiese

**Comment by creepystefan** [comment]
Über die TX und RX.
Du brauchst dazu einen USB zu TTL/UART rs232 Adapter. Werde noch bei meinem anderen Repo speziell für den Sinilink Adapter ne kleine Anleitung dazu schreiben.
Sollte ich diese awocje noch hinbekommen.
Heute leider nicht mehr.
Das bekommen wir schon hin.

**Reply by Tiese** [reply]
Ah.

Geht sowas auch: [USB to ESP8266 Wireless Transceiver Modul Programmer WLAN WiFi USB-Adapter CH340](https://www.ebay.de/itm/295806240394)?

**Reply by creepystefan** [reply]
Nehmen kannst du dafür viele.
Z.B. sowas ähnliches benutze ich.

https://www.az-delivery.de/products/hw-598-usb-auf-seriell-adapter-mit-cp2102-chip-und-kabel?variant=18282766794848&utm_source=google&utm_medium=cpc&utm_campaign=19229855661&utm_content=147170319769&utm_term=&gad_source=1&gclid=EAIaIQobChMI8tLt4MDThgMVZaSDBx1XdQ4sEAQYASABEgJFcfD_BwE

Oder 

https://www.ebay.de/itm/265212430374?chn=ps&_ul=DE&_trkparms=ispr%3D1&amdata=enc%3A17sr3ehs9TYyWFFYNIXndMQ99&norover=1&mkevt=1&mkrid=707-134425-41852-0&mkcid=2&mkscid=101&itemid=265212430374&targetid=1716911581439&device=m&mktype=pla&googleloc=1004102&poi=&campaignid=17943303986&mkgroupid=140642150118&rlsatarget=pla-1716911581439&abcId=9301060&merchantid=494547907&gad_source=1&gbraid=0AAAAAD_G4xZh6w2p8EICEwLn_EGYbDncR&gclid=EAIaIQobChMInY2HlMHThgMV-JCDBx3mDgetEAQYBSABEgIqEPD_BwE

Würde auch gehen

Wenn ich daheim bin ist es mit Laptop leichter.
Dann schreibe ich was in meinem anderen Repo (Sinilink) wie man es bespielt

Allerdings ist die Sknilink Programmierung weg. Dann gibt es keine Verbindung mehr über dieses sinilink Portal.

Hast du Homeassistent oder was ähnliches zum programmieren von ESP Chips?

**Comment by Tiese** [comment]
> Nehmen kannst du dafür viele.

Gut, den von mir genannten habe ich hier.

<br>

> Allerdings ist die Sknilink Programmierung weg. Dann gibt es keine Verbindung mehr über dieses sinilink Portal.

Das funktioniert sowieso nicht mit einem iPad! :-(
<br>

> Hast du Homeassistent oder was ähnliches zum programmieren von ESP Chips?

Ich habe einen ESP32 mit Arduino IDE programmiert.
Dieser limitiert mir meinen Wechselrichter und schaltet diesen EIN/AUS, je nach Batteriestand.
Jetzt möchte ich zudem gerne noch den XY6020L ebenfalls darüber EIN/AUS schalten können.

Falls das möglich ist?

**Comment by creepystefan** [comment]
Ja das klappt. Bei mir ladet es meine Batterie gesteuert über die Leistung wie viel ich über einspeise.
Die Programmierung wäre dann etwas anders. Aber nicht EinAus sondern leistungsgesteuert.
Bedenke die Eingangsspannung sollte immer 5v höher sein als am Ausgang erforderlich.
Was hat dein Eingang? Bei 24v wird es kritisch weil die meisten WR erst bei 22v einschalten.

Dann brauchst du noch ESPHome gibts als Python Variante.

**Comment by Tiese** [comment]
Nein, so ist das bei mir nicht.
Ich habe eine 48V-Batterie. Der XY6020L lädt diese nicht, sondern versorgt den angeschlossenen Wechselrichter mit Spannung.
Da dieser sich bereits bestens regelt und ausschaltet (wenn die Batterie leer ist - durch meinen ESP32), brauche ich nur noch den XY6020L über dessen WiFi ausschalten (noch etwas Strom sparen). Wenn dann die Batterie wieder durch den Victron SmartSolar geladen wurde, erhält der XY6020L sein Einschaltsignal, damit der Wechselrichter wieder seine Arbeit aufnehmen kann.

So ist es zumindest abgedacht - nix mit Home Assistent oder ESPHome?!

**Comment by creepystefan** [comment]
Achso du willst nur wissen wie du das Sinilink programmieren kannst.
Weil bei mir läuft die Programmierung nicht über arduino
Sondern über esphome.
Ist auch eine Programmiersprache (yaml configurier) datei.

Wenn du meinen Code benutzen möchtest wäre es ESPhome.

Mit arduino bin ich raus.

Kann dir nur mit EspHome dienen.

Du kannst auch den Strom regeln.
Dann brauchst du den WR nicht mehr regeln (begrenzen)

**Comment by Tiese** [comment]
> Wenn du meinen Code benutzen möchtest wäre es ESPhome.

Ist denn dazu Home Assistent nötig oder kann ich dann den Sinilink-WiFi auch so abfragen?

<br>

> Du kannst auch den Strom regeln.

Das wäre jetzt noch OK, aber nicht mehr, wenn ich einen 2. Wechselrichter dazu schalten möchte.

**Reply by Tiese** [reply]
Braucht einen weiteren ESP32 worauf dann ESPHome läuft?

**Comment by Tiese** [comment]
Ich bin gerade sehr verwirrt. &#129327;

Muss auf den Sinilink "**xy6020-esp8266-control**" oder "**Tasmota**" geflasht werden, um ihn zu steuern?

Für diese Info wäre ich sehr dankbar!

**Comment by creepystefan** [comment]
Ahh Achso. Also ESPHome ist eher so wie Arduino.
Man braucht keinen Homeassist.
Für dich müsste ich ein komplett neues Programm erstellen.
Ich habe es bei mir so programmiert dass der ESP komplett unabhängig von irgendwas anderem regelt. Natürlich mit Anfrage über einen Shelly im Sicherungskasten.
Hast du bestimmt auch?
ESPHome arbeitet nur sehr gut mit Hlmeassist zusammen. Ist nicht nötig.
Aktuell kann man mit dem Sinilink nur LED auf dem Board schalten bzw. Ausgänge schalten. 
Weiter bin ich noch nicht.
Am besten wäre es wenn du mal ein Test file nur auf das WLAN Modul programmierst.
Nur um zu schauen ob es geht.
ESP Home ist ähnlich wie Tasmota.
Mit meinem Controller kannst kannst du den XY6020 steuern über WLAN so wie die Sinilink App.
Die Regelung wie es steuern soll ist darin nicht enthalten.

Ich hoffe du kommst mit

**Comment by Tiese** [comment]
> Man braucht keinen Homeassist.

Das ist schon mal gut. 👍 

<br>
Jetzt weiß ich aber immer noch nicht, was genau auf den Sinilink drauf muss, um diesen überhaupt ansprechen zu können?

**Comment by creepystefan** [comment]
Also entweder das XY6020.yaml
Dann brauchst du aber irgendwas dass den XY6020 regelt nach deinen Bedürfnissen.
Oder
Wir programmieren dir eine eigene Lösung speziell für dich.
Erst einmal müssen wir allgemein ein ESPhome Programm auf den Sinilink bekommen.
Dazu brauchst du erstmal den ESPhome installier. Das geht auch per Web Seite.
Verbinde mal den Sinilink mit 5v GND und RX und TX mit dem usb-ttl Adapter. 
Dann gehst du auf ESPhome WEB installer. Und programmierst deinen Sinilink. Da gibt es ein standart Programm ob es überhaupt funktioniert.
Beim Programmieren musst du den IO0 und GND zusammen machen.
Während des Betriebs dann nicht mehr. Sonst startet er nicht.
Bekommst du das erstmal hin?
Du musst ja dann auch IP Adresse und dein WLAN vergeben.

**Reply by Tiese** [reply]
Vielen Dank, das werde ich versuchen!

**Comment by creepystefan** [comment]
Was hast du im Sicherungskasten dran dass du deine Leistungen abfragen kannst?
Smart Meter? 
Z.b. Shelly 3em gen2?

**Reply by Tiese** [reply]
Einen Hichi.

**Comment by creepystefan** [comment]
So einen IR Zähler?
Wie wird das ins wlan übertragen?
Tasmota? Gibt es da eine API dazu?
Das muss der ESP ja abfragen dann.
Aber der erkannt ja nur ob Strom verbraucht wird oder?
Aber keine Einspeisung?
Liege ich da richtig? Oder ist es ein anderer Hichi?
Dann wird es schwer mit Strom Regelung.
Dann ist es echt nur ein aus schalten.
Ich werde morgen Abend mal eine Zeichnung rein setzen.

**Reply by Tiese** [reply]
> So einen IR Zähler?

Ja.

>Wie wird das ins wlan übertragen?

Ganz normal über den Aufruf der IP-Adresse.

>Tasmota?

Ja.

>Gibt es da eine API dazu?

HTTP(S) + JSON Konfiguration

>Aber der erkannt ja nur ob Strom verbraucht wird oder?
Aber keine Einspeisung?

Verbrauch z.B. 100 W - Einspeisung z.B. **-100 W**

**Comment by creepystefan** [comment]
Perfekt. Dann schauen wir erstmal dass wir ESPhome auf dem sinilink zum laufen bekommen der Rest wird dann eher leicht.Gib Bescheid.

**Reply by Tiese** [reply]
Mach ich.

Vielen Dank für deine schnelle und gute Hilfe! 👍

**Comment by Tiese** [comment]
So weit bin ich gekommen...
Wenn ich IO0 und RST brücke, blinkt die LED vom Sinilink sehr schnell.

Starte ich in ESPHome (web) die Installation, erscheint das hier...
![Bildschirmfoto 2024-06-12 um 08 42 27](https://github.com/creepystefan/esphome-XY6020/assets/141029037/f3a2b2ba-1bc4-486f-b52f-2b20ad2be262)
Und das bleibt ewig - ohne Änderung :-(

**Comment by creepystefan** [comment]
Nicht RST
sondern GND 
Also IO0 und GND
Verbinden.
Dann Strom drauf geben.
Flashen / installieren
Wenn fertig.
Dann Strom weg.
IO0 von Gnd trennen.
Dann Strom anschliesen
Dann sollte er über die IP Adresse erreichbar sein

**Reply by Tiese** [reply]
Ich depp :-/

Jetzt hat es geklappt :-)

**_Vielen Dank!_**

**Comment by creepystefan** [comment]
Sehr gut. Kein Stress.
In der FRITZ!Box müsste jetzt die IP ersichtlich sein.
Damit kommst du auf die Web Oberfläche. Geht das?
Wenn man den ESP flashed steht in den Logs ein API und OTA key.
Den AP key brauchst eher nicht der ist für homeassist.
API und OTA key notieren.
Hier nicht nennen. Ist dein Key für dein Board. 
Ab 12 Uhr ist es bei mir bis abends schwierig.
Es gibt viele YouTube Videos wegen dem OTA und API Key.
Ich habe einen neuen Ordner erstellt. Name Test.
Darin befindet sich eine secret.yaml
Bitte herunterladen und die Daten anpassen.
Die eingetragenen Daten nicht nennen. Brauche ich nicht. Nur du.
Dann befasse dich mal mit esphome Windows installieren.
Brauchst Python. (Ist wie arduino IDE)
Dann noch unter Python eSPhome installieren.
Wenn installiert dann auch diese secret.yaml in dem installierten esphome und ergänze es mit meinem secret.yaml.
Wenn du das alles hast.
Kannst du mal im Ordner Test die Test.yaml flashen.

Ich hoffe es wird jetzt nicht zu viel.
Es gibt viele Videos über esphome.
Die reine programmierung mach ich dann.

**Comment by creepystefan** [comment]
Wenn du es geschafft hast meine Test.yaml zu flashen.
Dann solltest du unter der IP Adresse mit Port 9005
Also 19.168.178.22:9005 die Webseite öffnen können.
Ipadresse vom Sinilink esp
Ip:9005
Dann sollten da daten stehen
Power 1-6
Nenne mir dann bitte die Zahl.
Und ob es mit dem Ichi smart Meter übereinstimmt.
Dann kann ich weiter machen.

**Reply by creepystefan** [reply]
Ich höre garnix mehr von dir, hast du schon weiter gemacht?

**Reply by Tiese** [reply]
hm...
die Startseite sieht ziemlich leer aus...
![Bildschirmfoto 2024-06-12 um 16 59 51](https://github.com/creepystefan/esphome-XY6020/assets/141029037/18f5ed94-ab2c-4178-952e-e94d21f746a5)

Irgendwelche Keys kann ich nicht finden :-(
Und die Logs sind leer?!

**Reply by Tiese** [reply]
> esphome Windows installieren

Ich habe eine MacMini.

**Comment by creepystefan** [comment]
Gibt es auch für Mac.

https://esphome.io/guides/installing_esphome.html

Eine andere Möglichkeit wäre dass du mir dein Wifi Zugang nennst
Und die IP Adresse die der ESP aktuell hat.
Oder du schreibst mir eine email?
Dann kann ich dir eine .bin Datei erstellen.
